examples/k-means/k-means.hpp

```diff
@@ -224,11 +225,96 @@ auto KMeans(const DIA<Point, InStack>& input_points,
                      return cc.center.p / static_cast<double>(cc.center.count);
                  })
             .Collapse();
+
+		// Check whether centroid positions changed significantly,
+		// if yes do another iteration
+		break_condition = true;
+		local_centroids = centroids.AllGather();
+		for (size_t i = 0; i < local_centroids.size(); ++i) {
+			if (local_centroids[i].Distance(old_local_centroids[i]) > epsilon) {
+				old_local_centroids.insert(old_local_centroids.begin(),
```

Comment:
why this insert() ? old_local_centroids is also not cleared. This cannot be correct.